pkg/workloadapi/workload.proto

```diff
@@ -17,7 +17,71 @@ syntax = "proto3";
+// Address is intended to be able to be looked up on-demand, allowing a client
+// to answer a question like "what is this IP address", similar to a reverse DNS lookup.
+//
+// Each resource will have a mesh-wide unique opaque name, defined in the individual messages.
+// In addition, to support lookup by IP address, they will have *alias* names for each IP the resource represents.
+// There may be multiple aliases for the same resource (examples: service in multiple networks, or a dual-stack workload).
+// Aliases are keyed by network/IP address. Example: "default/1.2.3.4".
+//
+// In some cases, we do not know the IP address of a Workload. For instance, we may simply know
+// that there is a workload behind a gateway, and rely on the gateway to handle the rest.
+// In this case, the key format will be "network/resource-uid". The resource can be a Pod, WorkloadEntry, etc.
+// These resources cannot be looked up on-demand.
+//
+// In some cases, we do not know the IP address of a Service. These services cannot be used for matching
+// outbound traffic, as we only have L4 attributes to route based on. However,
+// they can be used for Gateways.
+// In this case, the key format will be "network/hostname".
+// These resources cannot be looked up on-demand.
+message Address {
+  oneof type {
+    // Workload represents an individual workload.
+    // This could be a single Pod, a VM instance, etc.
+    Workload workload = 1;
+    // Service represents a service - a group of workloads that can be accessed together.
+    Service service = 2;
+  }
+}
+
+// Service represents a service - a group of workloads that can be accessed together.
+// The primary key is "namespace/hostname".
+// Secondary (alias) keys are the unique `network/IP` pairs that the service can be reached at.
+message Service {
+  // Name represents the name for the service.
+  // For Kubernetes, this is the Service name.
+  string name = 1;
+  // Namespace represents the namespace for the service.
+  string namespace = 2;
+  // Hostname represents the FQDN of the service.
+  // For Kubernetes, this would be <name>.<namespace>.svc.<cluster domain>
+  // TODO: support this field
+  string hostname = 3;
+  // Address represents the addresses the service can be reached at.
+  // There may be multiple addresses for a single service if it resides in multiple networks,
+  // multiple clusters, and/or if it's dual stack (TODO: support dual stack).
+  repeated NetworkAddress addresses = 4;
+  // Ports for the service.
+  // The target_port may be overridden on a per-workload basis.
+  repeated Port ports = 5;
+  // Optional; if set, the SAN to verify for TLS connections.
+  // Typically, this is not set and per-workload identity is used to verfiy
+  // TODO: support this field
+  repeated string subject_alt_names = 6;
+}
+
+message OpaqueEndpoint {
+  // Tunneling protocol to use when connecting to the service, if any.
+  TunnelProtocol tunnel_protocol = 1;
+}
```

Comment:
we need to update the proto to latest and remove this as its no longer needed

Reply:
I thought I copied your .proto file over from ztunnel and re-ran the make gen, must have copied the wrong version